accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -325,6 +354,30 @@ not a subtype of the class type associated with the class in which it is defined
 (specifically, it is an error to return a nullable type from a factory
 constructor for any class other than `Null`).
 
+It is an error if any case of a switch statement except the last case (the
+default case if present) **may complete normally**.  The previous syntactic
+restriction requiring the last statement of each case to be one of a delimited
+set of statements (break, continue, return, throw, or rethrow) is removed.
+
+Given a switch statement which switches over an expression `e` of type `T`,
+where the cases are dispatched based on expressions `e0...ek`:
+  - It is no longer required that the `ei` be instances of the same class.
+  - It is an error if the static type of any of the `ei` is not a subtype of
+    `T`.
```

Comment:
We rarely use the static type of constant expressions for anything. Use the runtime type instead.

Reply:
Hmm.  It worries me slightly.   If it wasn't for the potential for additional breakage, I think I'd be mildly opposed on the grounds that it ties our hands later for no good reason.  If we do generalize to patterns later, patterns will mostly be driven off of static types.  But I can't immediately think of anything that goes too wrong with just treating *references* to constants as equivalent to the constants themselves, and just ignoring their static types.

Reply:
I guess either should work. The only case where it might cause trouble is if we have a const variable with a more general type than the value itself, which is likely incredibly rare, and the switch expression has a more specific type. I can't imagine a use-case where that happens.